plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/DigestExtractor.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+import org.sonatype.nexus.blobstore.api.Blob;
+import org.sonatype.nexus.common.io.LimitedInputStream;
+import org.sonatype.nexus.repository.view.Payload;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Strings;
+import com.google.common.io.CharStreams;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Utility class for extracting digest strings from Maven .sha1/.md5 files that originates from external source, and
+ * might be created by vastly different tools out there.
+ *
+ * @since 3.0
+ */
+public class DigestExtractor
+{
+  private static final int MAX_CHARS_NEEDED = 120;
+
+  private DigestExtractor() {}
+
+  /**
+   * Extract the digest string from a payload, that carries content of a Maven sha1/md5 file.
+   *
+   * @see {@link #extract(String)}
+   */
+  @Nullable
+  public static String extract(final Payload payload)
+      throws IOException
+  {
+    checkNotNull(payload);
+    try (InputStream is = payload.openInputStream()) {
+      return extract(is);
+    }
+  }
+
+  /**
+   * Extract the digest string from a blob, that carries content of a Maven sha1/md5 file.
+   *
+   * @see {@link #extract(String)}
+   */
+  @Nullable
+  public static String extract(final Blob blob)
+      throws IOException
+  {
+    checkNotNull(blob);
+    try (InputStream is = blob.getInputStream()) {
+      return extract(is);
+    }
+  }
+
+  /**
+   * Extract the digest string from a stream, that carries a payload of a Maven sha1/md5 file.
+   *
+   * @see {@link #extract(String)}
+   */
+  @Nullable
+  public static String extract(final InputStream stream)
+      throws IOException
+  {
+    checkNotNull(stream);
+    try (InputStreamReader isr = new InputStreamReader(
+        new LimitedInputStream(stream, 0, MAX_CHARS_NEEDED), Charsets.UTF_8)) {
+      return extract(CharStreams.toString(isr));
+    }
+  }
```

Comment:
This class is in internal package (like everything else of maven stuff so far is), is not public API. Will lessen class visibility if that's enough.